src/control/lib/support/log.go

```diff
@@ -579,6 +659,90 @@ func copyServerConfig(log logging.Logger, opts ...CollectLogsParams) error {
 	return nil
 }
 
+// Copy only specific lines from the server logs based on the Start/End date
+// and Start/End time, provided by user.
+func cpLinesFromLog(log logging.Logger, srcFile string, destFile string, opts ...CollectLogsParams) error {
+	// Copy the full log file in case of no dates provided
+	if opts[0].LogStartDate == "" && opts[0].LogEndDate == "" {
+		err := cpLogFile(srcFile, destFile, log)
+		if err != nil {
+			return err
+		}
+	} else {
+		var startTime, endTime string
+
+		logStartDate, err := time.Parse(MMDD, opts[0].LogStartDate)
+		if err != nil {
+			return err
+		}
+
+		logEndDate, err := time.Parse(MMDD, opts[0].LogEndDate)
+		if err != nil {
+			return err
+		}
+
+		// Default Start time, in case no start time provides on start dates.This will copy log start of the day.
+		if opts[0].LogStartTime == "" {
+			opts[0].LogStartTime = "00:00:00"
+		}
+
+		// Default End time, in case no End time provides.This will copy log till the End of the day.
+		if opts[0].LogEndTime == "" {
+			opts[0].LogEndTime = "23:59:59"
+		}
+
+		startTime = fmt.Sprintf("%d/%d-%s", logStartDate.Month(), logStartDate.Day(), opts[0].LogStartTime)
+		endTime = fmt.Sprintf("%d/%d-%s", logEndDate.Month(), logEndDate.Day(), opts[0].LogEndTime)
+
+		actStartTime, _ := time.Parse(MMDDHHMMSS, startTime)
+		actEndTime, _ := time.Parse(MMDDHHMMSS, endTime)
+
+		// Create the new empty file, which will be used to copy the matching date/time log lines.
+		logFileName := filepath.Join(destFile, filepath.Base(srcFile))
+		writeFile, err := os.Create(logFileName)
+		if err != nil {
+			return err
+		}
+		defer writeFile.Close()
+
+		readFile, err := os.Open(srcFile)
+		if err != nil {
+			return err
+		}
+		defer readFile.Close()
+
+		// Loop through each line and identify the date and time of each log line.
+		// Compare the date,time stamp and copy if it's in the user provided range.
+		scanner := bufio.NewScanner(readFile)
+		for scanner.Scan() {
+			lineData := scanner.Text()
+			lineDataSlice := strings.Split(lineData, " ")
+			dateTime := strings.Split(lineDataSlice[0], "-")
+			// skip and continue to next line if no date time stamp
+			if len(dateTime) != 2 {
+				continue
+			}
```

Comment:
Not sure if I agree with this. I think it makes sense to copy everything between the start and end date/time, even if it's formatted strangely.

Maybe you could set a bool when you see the start time and break out of the loop when you see the end time? Then you would check the flag to decide whether to write the line.

Reply:
Yes we can do that as well. When i tested this, only first line is the one which has extra line and rest all logging is fine and has individual log lines with date/time stamps.

Reply:
I think it's a good idea to ensure we can handle multi-line log entries as well. I know we do them on the client side.

Reply:
ok I will update the logic for multiple lines which doesn't have the date/time stamp.